contentful/apps
# Improve error logging in Slack app and fix bug

## Purpose

This PR adjusts the following: 

1. Adds more meaningful messages and identifiers to errors thrown within the Slack app lambda. 
2. Fixes [this](https://contentful.atlassian.net/browse/INTEG-652) issue. 

Also as you will notice, lots of trailing commas are added as per the prettier commit hook. I am not sure if this is standard, but I have trouble reverting the changes locally given the commit hook does them, though I could via github if they are an issue. Please let me know your thoughts!

## Approach

1. This PR adjusts the `Exception` class within the API to handle the addition of identifying information like `environmentId` and `spaceId`. Additionally, more meaningful error messages are provided when the errors are manually thrown. 
2. This was a small issue where a snackbar was rendering and indicating to the user that an error occurred, despite successful installation. This snackbar is being initialized from the `useConnect` component which is the popup that renders when connecting to Slack via the auth flow. The error being thrown that is causing this snackbar is simply a race condition of the API grabbing installation parameters via the `CMA` just a little before the app is actually labeled as "installed". I believe the only reason this is occurring is because the `EventsService` class, that fetches these installation parameters on subsequent requests, is being instantiated when the auth token is created (`POST` call to `/api/tokens`) in the `useConnect` popup auth flow. The error is not meaningful and would require a bit of an overhaul to completely prevent it from being thrown (as far as I understand, but please let me know if you believe otherwise!), so for now to improve user experience, this PR just renders an error snackbar if the error is anything other than this specific race condition error. 

## Testing steps

Uninstall and install the Slack app in your local environment, and confirm that all works correctly and no extraneous error messages appear.
